### ISSUE TITLE
shouldReloadAll will change in Ember Data 2.0

### ISSUE THREAD

**Issue by broerse** [nolanlawson/ember-pouch#79]
I get this deprecation warning:
`Ember Inspector (Deprecation Trace): The default behavior of shouldReloadAll will change in Ember Data 2.0 to always return false when there is at least one "author" record in the store. If you would like to preserve the current behavior please override shouldReloadAll in your adapter:application and return true.`
Do we need to change something? Is the new behavior better?

**Comment by fsmanuel** [comment]
The new behavior aims to optimize performance.
There are 4 methods:
- [shouldReloadRecord](http://emberjs.com/api/data/classes/DS.RESTAdapter.html#method_shouldReloadRecord)
- [shouldBackgroundReloadRecord](http://emberjs.com/api/data/classes/DS.RESTAdapter.html#method_shouldBackgroundReloadRecord)
- [shouldReloadAll](http://emberjs.com/api/data/classes/DS.RESTAdapter.html#method_shouldReloadAll)
- [shouldBackgroundReloadAll](http://emberjs.com/api/data/classes/DS.RESTAdapter.html#method_shouldBackgroundReloadAll)

The idea in Ember Data 2.0 is to return a cached version immediately and than refetch in the background.

For pouchdb the fetch isn't expensive because we don't need an ajax request. I guess it's best to always reload:

```js
  // Reload behavior
  shouldReloadRecord: function() { return true; },
  shouldReloadAll: function() { return true; },
  shouldBackgroundReloadRecord: function() { return true; },
  shouldBackgroundReloadAll: function() { return true; }
```

**Comment by broerse** [comment]
Perhaps we should implement this in the ember-pouch adapter by default.

**Comment by fsmanuel** [comment]
:+1:

**Comment by broerse** [comment]
I will write a PR for this.

**Comment by rsutphin** [comment]
The change watcher already provides automatic refreshing of cached records in a way that's more efficient than this ember-data feature — it updates the store-cached record as soon as it is changed (not just when `findRecord` is called), and it only updates when it is actually different (instead of making a query every time `findRecord` is called). So I think returning true from `shouldReloadRecord` or `shouldBackgroundReloadRecord` in an ember-pouch app will make the app do more work without benefit. I'd be in favor of having ember-pouch default both of those to `false`.

`shouldReloadAll` is trickier and requires more thought. It's easy to say that, since an ember-pouch query is local, it's cheap and therefore it doesn't matter if we do it all the time. But when you have a lot of records of a particular type, it's still slow to deserialize them all. Doing this frequently will make your app noticeably laggy, even if there are no changes. So I'm not sure what the proper default is for this. For the moment, I think it's worth allowing folks who are using ember-data 1.13 to see the deprecation warning and decide for themselves.

**Comment by broerse** [comment]
@rsutphin Oh, just now saw your comments. I still think enabling the new Reload some time after 2.0 is released is better. .0 versions are not always OK ;-)

**Comment by rsutphin** [comment]
Addressed by #83.

**Comment by broerse** [comment]
Thanks!
